io/net/multiport_stress.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>
+
+import netifaces
+from avocado import main
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+
+
+class MultiportStress(Test):
+    '''
+    To perform IO stress on multiple ports on a NIC adapter
+    '''
+
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        self.host_interfaces = self.params.get("host_interfaces",
+                                               default="").split(",")
+        if not self.host_interfaces:
+            self.skip("user should specify host interfaces")
+        pkg = "iputils-ping"
+        smm = SoftwareManager()
+        if not smm.check_installed(pkg) and not smm.install(pkg):
+            self.skip("%s package is need to test" % pkg)
+        self.peer_ips = self.params.get("peer_ips",
+                                        default="").split(",")
+        interfaces = netifaces.interfaces()
+        for self.host_interface in self.host_interfaces:
+            if self.host_interface not in interfaces:
+                self.skip("interface is not available")
+        self.count = self.params.get("count", default="1000")
+
+    def test_multiport_stress(self):
+        '''
+        Ping to multiple peers parallely
+        '''
+        parallel_procs = []
+        for host, peer in map(None, self.host_interfaces, self.peer_ips):
+            self.log.info('Starting Ping test')
+            cmd = "ping -I %s %s -c %s" % (host, peer, self.count)
```

Comment:
can only way to run ping to generate stress  can we not run netperf and iperf test ?

@narasimhan-v any thought about this ?